src/main/java/entity/Task.java

```diff
@@ -4,21 +4,28 @@
 import java.time.Duration;
 import java.time.LocalDateTime;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 /**
  * A Task stores the name of the task, whether it is completed,
  * the time needed to complete the task (optional), the deadline
  * of the class (optional), and subtasks (optional)
+ *
+ * - timeNeeded: the time duration needed to complete the task
+ * - notificationTimes: a set of times in Duration presenting how long in advance
+ *                      to send the event remainder notification
  */
 public class Task {
 
+    private final int id;
```

Comment:
In the future try avoiding making small changes like this that aren't functionally relevant to the PR. Small refactoring changes should be put into small refactoring PRs

Reply:
Should I rollback the changes for this part to make this pr clearer?

Reply:
Ah no sorry you don't need to. It's just something to take into consideration moving forward because it makes merge conflicts easy to occur. E.g. I think this might create conflict with the Id PR because that is changing the type of this attribute to long.

Reply:
Ah ok sorry about that. We can wait for the id pr to get merged and I'll resolve any conflicts before merging this pr into main.

Reply:
Ok I'll leave it as this for now and resolve any conflicts in the future